dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/TokenFilter.java

```diff
@@ -47,10 +47,14 @@ public Result invoke(Invoker<?> invoker, Invocation inv)
             Class<?> serviceType = invoker.getInterface();
             Map<String, Object> attachments = inv.getObjectAttachments();
             String remoteToken = (attachments == null ? null : (String) attachments.get(TOKEN_KEY));
-            if (!token.equals(remoteToken)) {
-                throw new RpcException("Invalid token! Forbid invoke remote service " + serviceType + " method " + inv.getMethodName() +
-                        "() from consumer " + RpcContext.getServiceContext().getRemoteHost() + " to provider " +
-                        RpcContext.getServiceContext().getLocalHost()+ ", consumer incorrect token is " + remoteToken);
+            if (ConfigUtils.isDefault(token)){
+                inv.setAttachment(TOKEN_KEY,token);
+            }
+            if (!token.equals(remoteToken) && !ConfigUtils.isDefault(token)) {
```

Comment:
This may cause token downgrade attack. If provider use default token, TokenFilter in provider side may not work.

Reply:
> This may cause token downgrade attack. If provider use default token, TokenFilter in provider side may not work.

@AlbumenJ You mean that when the token is passed in the thread context it should fake a value, or expose a configuration. Is it used to guarantee idempotency when downgrading?